Switch from WTF::Span to std::span

ddkilzer opened the pull request:
#### 6d41b0f74b61f318f8f50a191f4bb5808a507e43
<pre>
Switch from WTF::Span to std::span
<a href="https://bugs.webkit.org/show_bug.cgi?id=253377">https://bugs.webkit.org/show_bug.cgi?id=253377</a>
&lt;rdar://103182220&gt;

Reviewed by Darin Adler.

On Apple ports of WebKit, switch to use std::span instead of
WTF::Span by creating a template type alias, creating
WTF::makeSpan() to use for both types (since the type alias
can&apos;t perform deduction), and implementing wrappers for
WTF::asBytes() and WTF::asWritableBytes().

* Source/JavaScriptCore/jsc.cpp:
(JSC_DEFINE_HOST_FUNCTION):
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/SHA1.cpp:
- Include &lt;cstddef&gt; for std::to_integer().
* Source/WTF/wtf/SHA1.h:
(WTF::SHA1::addBytes):
* Source/WTF/wtf/Span.h:
(WTF::makeSpan):
(WTF::asBytes):
(WTF::asWritableBytes):
(makeSpan):
* Source/WTF/wtf/persistence/PersistentDecoder.cpp:
(WTF::Persistence::Decoder::bufferIsLargeEnoughToContain const):
(WTF::Persistence::Decoder::bufferPointerForDirectRead):
(WTF::Persistence::Decoder::rewind):
(WTF::Persistence::Decoder::decodeNumber):
* Source/WTF/wtf/persistence/PersistentDecoder.h:
(WTF::Persistence::Decoder::currentOffset const):
- Switch to use a span iterator instead of a raw pointer for
  bounds checks.
(): Deleted.
* Source/WTF/wtf/text/Base64.cpp:
(WTF::base64EncodeInternal):
(WTF::base64Decode):
* Source/WTF/wtf/text/Base64.h:
(WTF::StringTypeAdapter&lt;Base64Specification&gt;::writeTo const):
* Source/WTF/wtf/text/StringBuilder.h:
(WTF::StringBuilder::span const):
* Source/WTF/wtf/text/StringImpl.h:
(WTF::StringImpl::tryGetUTF8ForCharacters):
* Source/WTF/wtf/text/StringSearch.h:
(WTF::BoyerMooreHorspoolTable::BoyerMooreHorspoolTable):
* Source/WTF/wtf/text/WTFString.h:
(WTF::String::tryGetUTF8 const):

* Source/WebCore/Modules/fetch/FetchResponse.cpp:
(WebCore::FetchResponse::BodyLoader::didReceiveData):
(WebCore::FetchResponse::BodyLoader::consumeDataByChunk):
* Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.cpp:
(WebCore::IDBServer::SQLiteIDBBackingStore::addExistingIndex):
(WebCore::IDBServer::SQLiteIDBBackingStore::createObjectStore):
(WebCore::IDBServer::SQLiteIDBBackingStore::createIndex):
(WebCore::IDBServer::SQLiteIDBBackingStore::uncheckedHasIndexRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::uncheckedPutIndexRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::keyExistsInObjectStore):
(WebCore::IDBServer::SQLiteIDBBackingStore::deleteRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::addRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::getRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::getAllObjectStoreRecords):
(WebCore::IDBServer::SQLiteIDBBackingStore::uncheckedGetIndexRecordForOneKey):
(WebCore::IDBServer::SQLiteIDBBackingStore::getCount):
* Source/WebCore/Modules/indexeddb/server/SQLiteIDBCursor.cpp:
(WebCore::IDBServer::SQLiteIDBCursor::bindArguments):
(WebCore::IDBServer::SQLiteIDBCursor::resetAndRebindPreIndexStatementIfNecessary):
* Source/WebCore/Modules/push-api/PushDatabase.cpp:
(WebCore::uuidToSpan):
* Source/WebCore/Modules/push-api/PushMessageCrypto.cpp:
(WebCore::PushCrypto::decryptAES128GCMPayload):
(WebCore::PushCrypto::decryptAESGCMPayload):
* Source/WebCore/Modules/push-api/cocoa/PushCryptoCocoa.cpp:
(WebCore::PushCrypto::decryptAES128GCM):
- Stop using span iterators as raw pointers.
* Source/WebCore/Modules/webauthn/AuthenticatorResponseData.h:
(WebCore::encodeArrayBuffer):
* Source/WebCore/Modules/webauthn/cbor/CBORReader.cpp:
(cbor::CBORReader::CBORReader):
(cbor::CBORReader::read):
(cbor::CBORReader::readString):
(cbor::CBORReader::readBytes):
(cbor::CBORReader::canConsume):
(cbor::CBORReader::checkExtraneousData):
(cbor::CBORReader::~CBORReader): Make default.
* Source/WebCore/Modules/webauthn/cbor/CBORReader.h:
- Stop using span iterators as raw pointers in CBORReader.
- Store reference to span instead of begin itertor since both
  begin and end iterators are needed for pointer math.
- Update class to use span iterators to calculate offsets.
* Source/WebCore/bindings/js/SerializedScriptValue.h:
(WebCore::SerializedScriptValue::encode const):
* Source/WebCore/css/CSSValueList.h:
- Include &lt;array&gt; since it&apos;s no longer included by &lt;wtf/Span.h&gt;.
* Source/WebCore/dom/make_names.pl:
(printTagNameCppFile):
(printElementNameCppFile):
* Source/WebCore/fileapi/NetworkSendQueue.cpp:
(WebCore::NetworkSendQueue::enqueue):
(WebCore::NetworkSendQueue::processMessages):
* Source/WebCore/html/canvas/CanvasPath.cpp:
(WebCore::CanvasPath::roundRect):
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::HTMLFastPathParser::scanText):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::TextControlInnerTextElement::updateInnerTextElementEditabilityImpl):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::ApplicationCacheStorage::store):
* Source/WebCore/loader/icon/IconLoader.cpp:
(WebCore::IconLoader::notifyFinished):
* Source/WebCore/page/PageColorSampler.cpp:
(WebCore::PageColorSampler::sampleTop):
* Source/WebCore/platform/SharedBuffer.h:
(WebCore::SharedBuffer::dataAsSpanForContiguousData const):
- Add RELEASE_ASSERT() to make sure the data is contiguous when
  creating a span, because it&apos;s a bug if not.
- Do not make SharedBuffer a span container since not every
  instance holds a contiguous buffer.
- Instead use dataAsSpanForContiguousData() to create a span.
* Source/WebCore/platform/graphics/ByteArrayPixelBuffer.h:
(WebCore::ByteArrayPixelBuffer::encode const):
* Source/WebCore/platform/gstreamer/GStreamerElementHarness.cpp:
(WebCore::MermaidBuilder::span):
* Source/WebCore/platform/sql/SQLiteStatement.cpp:
(WebCore::SQLiteStatement::bindBlob):
* Source/WebCore/platform/text/TextFlags.cpp:
(WebCore::computeFeatureSettingsFromVariants):
- Use Vector::span() to match return value of lambda.
* Source/WebCore/workers/service/server/RegistrationDatabase.cpp:
(WebCore::RegistrationDatabase::doPushChangesWithOpenDatabase):
* Source/WebCore/xml/XMLErrors.cpp:
(WebCore::createXHTMLParserErrorHeader):
(WebCore::XMLErrors::insertErrorMessageBlock):

* Source/WebKit/NetworkProcess/cache/CacheStorageEngine.cpp:
(WebKit::CacheStorage::Engine::writeSizeFile):
* Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
(WebKit::readRecordInfoFromFileData):
(WebKit::CacheStorageDiskStore::readRecordFromFileData):
(WebKit::CacheStorageDiskStore::writeRecords):
* Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
(WebKit::writeCachesList):
(WebKit::writeSizeFile):
* Source/WebKit/Platform/IPC/ArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;String&gt;::encode):
* Source/WebKit/Platform/IPC/ArgumentCoders.h:
(IPC::ArgumentCoder&lt;SHA1::Digest&gt;::encode):
* Source/WebKit/Platform/IPC/Encoder.h:
(IPC::Encoder::encodeObject):
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
(WebKit::SandboxExtension::createHandlesForMachLookup):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;WebCore::FragmentedSharedBuffer&gt;::encode):
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::encodeTypedArray):

* Source/WebKitLegacy/mac/History/BinaryPropertyList.cpp:
(IntegerArray::value_type): Add.
(IntegerArray::integers const): Delete.
(IntegerArray::data const): Rename from integers().
(add):
(IntegerArrayHash::equal):
- Update class to work as a span container.

* Source/cmake/OptionsCommon.cmake:
* Source/cmake/WebKitCompilerFlags.cmake:
- Define HAVE_STD_SPAN based on compiler test.  See also
  PlatformHave.h.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_wtf_make_span):
(check_style):
(CppChecker):
- Add checks for using &apos;Span { }&apos; and &apos;Span()&apos; to require
  makeSpan() instead.
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(WebKitStyleTest.test_wtf_make_span):
- Add tests for the new checks.
* Tools/TestWebKitAPI/Tests/IPC/ArgumentCoderTests.cpp:
(TestWebKitAPI::TYPED_TEST_P):
* Tools/TestWebKitAPI/Tests/WTF/EmbeddedFixedVector.cpp:
* Tools/TestWebKitAPI/Tests/WTF/RefCountedFixedVector.cpp:
- Include &lt;array&gt; since it&apos;s no longer included by &lt;wtf/Span.h&gt;.

Canonical link: <a href="https://commits.webkit.org/261239@main">https://commits.webkit.org/261239@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/be2d1d668e8e36c379e92c24b0f138bdac2235c7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110986 "21 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20125 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43553 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/2414 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119833 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/114945 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21501 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11227 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/2072 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116739 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/15973 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99153 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/103472 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/114409 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97930 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30817 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44407 "Passed tests") | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/99588 "Found 1 new JSC binary failure: testapi (failure)") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12643 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/32153 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/86316 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/10760 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13188 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9134 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/100711 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18603 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/51769 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31397 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7797 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15136 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/108748 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26799 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/Span.h:
```diff
@@ -7,8 +7,52 @@
+
+namespace WTF {
+
+inline constexpr std::size_t dynamic_extent = std::numeric_limits<std::size_t>::max();
```
Why not this?
```
using std::dynamic_extent;
```

ddkilzer replied:
Didn't realize that existed.  This line was copied from the `WTF::Span` section.  Will fix.

ddkilzer replied:
I didn't realize `WTF::dynamic_extent` was being used outside of the `Span.h` header, such as in `Source/WebKit/Platform/IPC/ArgumentCoders.h`:

<https://ews-build.webkit.org/#/builders/77/builds/20081>

Seems like I need this when enabling `std::span`:

`using dynamic_extent = std::dynamic_extent;`

ddkilzer replied:
Or rather...

`inline constexpr std::size_t dynamic_extent = std::dynamic_extent;`

darinadler commented on Source/WebCore/Modules/webauthn/cbor/CBORReader.h:
```diff
@@ -107,7 +107,7 @@ class CBORReader {
 
 private:
-    CBORReader(Bytes::iterator, const Bytes::iterator);
+    CBORReader(const Bytes&);
```
Should add `explicit` here.

darinadler commented on Tools/Scripts/webkitpy/style/checkers/cpp.py:
```diff
@@ -2815,6 +2815,44 @@ def check_ismainthread(filename, clean_lines, line_number, file_state, error):
 
 
+def check_wtf_make_span(clean_lines, line_number, file_state, error):
+    """Looks for use of 'Span { ... }' which should be replaced with 'WTF::makeSpan()'.
```
Don’t we also want to check for use of `std::span`? Would be good for portability to warn about that too.

ddkilzer replied:
Great point.  I was being a bit too forward-looking.

darinadler reviewed (approved):
Which non-Apple port still needs to use `WTF::Span`?

ddkilzer commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1371,6 +1371,11 @@
+// Only jsc-armv7 and jsc-mips bots do not support std::span.
+#if !(((CPU(ARM_THUMB2) && CPU(ARM_HARDFP)) || CPU(MIPS)) && OS(LINUX))
+#define HAVE_STD_SPAN 1
+#endif
```
This should probably not be necessary.

Not sure why jsc-armv7 and jsc-mips bots are not building with C++20.

ddkilzer replied:
I have a better solution by testing the GNU compiler via CMake to see if it has `<span>`.

Commit:
Switch from WTF::Span to std::span
https://bugs.webkit.org/show_bug.cgi?id=253377
<rdar://103182220>

Reviewed by Darin Adler.

On Apple ports of WebKit, switch to use std::span instead of
WTF::Span by creating a template type alias, creating
WTF::makeSpan() to use for both types (since the type alias
can't perform deduction), and implementing wrappers for
WTF::asBytes() and WTF::asWritableBytes().

* Source/JavaScriptCore/jsc.cpp:
(JSC_DEFINE_HOST_FUNCTION):
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/SHA1.cpp:
- Include <cstddef> for std::to_integer().
* Source/WTF/wtf/SHA1.h:
(WTF::SHA1::addBytes):
* Source/WTF/wtf/Span.h:
(WTF::makeSpan):
(WTF::asBytes):
(WTF::asWritableBytes):
(makeSpan):
* Source/WTF/wtf/persistence/PersistentDecoder.cpp:
(WTF::Persistence::Decoder::bufferIsLargeEnoughToContain const):
(WTF::Persistence::Decoder::bufferPointerForDirectRead):
(WTF::Persistence::Decoder::rewind):
(WTF::Persistence::Decoder::decodeNumber):
* Source/WTF/wtf/persistence/PersistentDecoder.h:
(WTF::Persistence::Decoder::currentOffset const):
- Switch to use a span iterator instead of a raw pointer for
  bounds checks.
(): Deleted.
* Source/WTF/wtf/text/Base64.cpp:
(WTF::base64EncodeInternal):
(WTF::base64Decode):
* Source/WTF/wtf/text/Base64.h:
(WTF::StringTypeAdapter<Base64Specification>::writeTo const):
* Source/WTF/wtf/text/StringBuilder.h:
(WTF::StringBuilder::span const):
* Source/WTF/wtf/text/StringImpl.h:
(WTF::StringImpl::tryGetUTF8ForCharacters):
* Source/WTF/wtf/text/StringSearch.h:
(WTF::BoyerMooreHorspoolTable::BoyerMooreHorspoolTable):
* Source/WTF/wtf/text/WTFString.h:
(WTF::String::tryGetUTF8 const):

* Source/WebCore/Modules/fetch/FetchResponse.cpp:
(WebCore::FetchResponse::BodyLoader::didReceiveData):
(WebCore::FetchResponse::BodyLoader::consumeDataByChunk):
* Source/WebCore/Modules/indexeddb/server/SQLiteIDBBackingStore.cpp:
(WebCore::IDBServer::SQLiteIDBBackingStore::addExistingIndex):
(WebCore::IDBServer::SQLiteIDBBackingStore::createObjectStore):
(WebCore::IDBServer::SQLiteIDBBackingStore::createIndex):
(WebCore::IDBServer::SQLiteIDBBackingStore::uncheckedHasIndexRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::uncheckedPutIndexRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::keyExistsInObjectStore):
(WebCore::IDBServer::SQLiteIDBBackingStore::deleteRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::addRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::getRecord):
(WebCore::IDBServer::SQLiteIDBBackingStore::getAllObjectStoreRecords):
(WebCore::IDBServer::SQLiteIDBBackingStore::uncheckedGetIndexRecordForOneKey):
(WebCore::IDBServer::SQLiteIDBBackingStore::getCount):
* Source/WebCore/Modules/indexeddb/server/SQLiteIDBCursor.cpp:
(WebCore::IDBServer::SQLiteIDBCursor::bindArguments):
(WebCore::IDBServer::SQLiteIDBCursor::resetAndRebindPreIndexStatementIfNecessary):
* Source/WebCore/Modules/push-api/PushDatabase.cpp:
(WebCore::uuidToSpan):
* Source/WebCore/Modules/push-api/PushMessageCrypto.cpp:
(WebCore::PushCrypto::decryptAES128GCMPayload):
(WebCore::PushCrypto::decryptAESGCMPayload):
* Source/WebCore/Modules/push-api/cocoa/PushCryptoCocoa.cpp:
(WebCore::PushCrypto::decryptAES128GCM):
- Stop using span iterators as raw pointers.
* Source/WebCore/Modules/webauthn/AuthenticatorResponseData.h:
(WebCore::encodeArrayBuffer):
* Source/WebCore/Modules/webauthn/cbor/CBORReader.cpp:
(cbor::CBORReader::CBORReader):
(cbor::CBORReader::read):
(cbor::CBORReader::readString):
(cbor::CBORReader::readBytes):
(cbor::CBORReader::canConsume):
(cbor::CBORReader::checkExtraneousData):
(cbor::CBORReader::~CBORReader): Make default.
* Source/WebCore/Modules/webauthn/cbor/CBORReader.h:
- Stop using span iterators as raw pointers in CBORReader.
- Store reference to span instead of begin itertor since both
  begin and end iterators are needed for pointer math.
- Update class to use span iterators to calculate offsets.
* Source/WebCore/bindings/js/SerializedScriptValue.h:
(WebCore::SerializedScriptValue::encode const):
* Source/WebCore/css/CSSValueList.h:
- Include <array> since it's no longer included by <wtf/Span.h>.
* Source/WebCore/dom/make_names.pl:
(printTagNameCppFile):
(printElementNameCppFile):
* Source/WebCore/fileapi/NetworkSendQueue.cpp:
(WebCore::NetworkSendQueue::enqueue):
(WebCore::NetworkSendQueue::processMessages):
* Source/WebCore/html/canvas/CanvasPath.cpp:
(WebCore::CanvasPath::roundRect):
* Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp:
(WebCore::HTMLFastPathParser::scanText):
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
(WebCore::TextControlInnerTextElement::updateInnerTextElementEditabilityImpl):
* Source/WebCore/loader/appcache/ApplicationCacheStorage.cpp:
(WebCore::ApplicationCacheStorage::store):
* Source/WebCore/loader/icon/IconLoader.cpp:
(WebCore::IconLoader::notifyFinished):
* Source/WebCore/page/PageColorSampler.cpp:
(WebCore::PageColorSampler::sampleTop):
* Source/WebCore/platform/SharedBuffer.h:
(WebCore::SharedBuffer::dataAsSpanForContiguousData const):
- Add RELEASE_ASSERT() to make sure the data is contiguous when
  creating a span, because it's a bug if not.
- Do not make SharedBuffer a span container since not every
  instance holds a contiguous buffer.
- Instead use dataAsSpanForContiguousData() to create a span.
* Source/WebCore/platform/graphics/ByteArrayPixelBuffer.h:
(WebCore::ByteArrayPixelBuffer::encode const):
* Source/WebCore/platform/gstreamer/GStreamerElementHarness.cpp:
(WebCore::MermaidBuilder::span):
* Source/WebCore/platform/sql/SQLiteStatement.cpp:
(WebCore::SQLiteStatement::bindBlob):
* Source/WebCore/platform/text/TextFlags.cpp:
(WebCore::computeFeatureSettingsFromVariants):
- Use Vector::span() to match return value of lambda.
* Source/WebCore/workers/service/server/RegistrationDatabase.cpp:
(WebCore::RegistrationDatabase::doPushChangesWithOpenDatabase):
* Source/WebCore/xml/XMLErrors.cpp:
(WebCore::createXHTMLParserErrorHeader):
(WebCore::XMLErrors::insertErrorMessageBlock):

* Source/WebKit/NetworkProcess/cache/CacheStorageEngine.cpp:
(WebKit::CacheStorage::Engine::writeSizeFile):
* Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp:
(WebKit::readRecordInfoFromFileData):
(WebKit::CacheStorageDiskStore::readRecordFromFileData):
(WebKit::CacheStorageDiskStore::writeRecords):
* Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
(WebKit::writeCachesList):
(WebKit::writeSizeFile):
* Source/WebKit/Platform/IPC/ArgumentCoders.cpp:
(IPC::ArgumentCoder<String>::encode):
* Source/WebKit/Platform/IPC/ArgumentCoders.h:
(IPC::ArgumentCoder<SHA1::Digest>::encode):
* Source/WebKit/Platform/IPC/Encoder.h:
(IPC::Encoder::encodeObject):
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
(WebKit::SandboxExtension::createHandlesForMachLookup):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<WebCore::FragmentedSharedBuffer>::encode):
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::encodeTypedArray):

* Source/WebKitLegacy/mac/History/BinaryPropertyList.cpp:
(IntegerArray::value_type): Add.
(IntegerArray::integers const): Delete.
(IntegerArray::data const): Rename from integers().
(add):
(IntegerArrayHash::equal):
- Update class to work as a span container.

* Source/cmake/OptionsCommon.cmake:
* Source/cmake/WebKitCompilerFlags.cmake:
- Define HAVE_STD_SPAN based on compiler test.  See also
  PlatformHave.h.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_wtf_make_span):
(check_style):
(CppChecker):
- Add checks for using 'Span { }' and 'Span()' to require
  makeSpan() instead.
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(WebKitStyleTest.test_wtf_make_span):
- Add tests for the new checks.
* Tools/TestWebKitAPI/Tests/IPC/ArgumentCoderTests.cpp:
(TestWebKitAPI::TYPED_TEST_P):
* Tools/TestWebKitAPI/Tests/WTF/EmbeddedFixedVector.cpp:
* Tools/TestWebKitAPI/Tests/WTF/RefCountedFixedVector.cpp:
- Include <array> since it's no longer included by <wtf/Span.h>.

Canonical link: https://commits.webkit.org/261239@main
